core/src/main/java/io/ultrabrew/metrics/data/BasicHistogramAggregator.java

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the Apache License 2.0 license. See LICENSE file in Ultrabrew Metrics
+// for terms.
+
+package io.ultrabrew.metrics.data;
+
+import static io.ultrabrew.metrics.Metric.DEFAULT_CARDINALITY;
+import static io.ultrabrew.metrics.Metric.DEFAULT_MAX_CARDINALITY;
+
+import io.ultrabrew.metrics.Histogram;
+import io.ultrabrew.metrics.util.DistributionBucket;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.Unsafe;
+
+public class BasicHistogramAggregator extends ConcurrentIntTable implements Aggregator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BasicHistogramAggregator.class);
+  private static final String[] AGG_FIELDS = {"count", "sum", "min", "max"};
+  private static final long[] IDENTITY = {0L, 0L, Long.MAX_VALUE, Long.MIN_VALUE};
+
+  private final String metricId;
+  private final DistributionBucket buckets;
+  private final int bucketCount;
+
+  public BasicHistogramAggregator(Histogram histogram) {
+    this(histogram.id, histogram.bucket, histogram.cardinality, histogram.maxCardinality);
+  }
+
+  public BasicHistogramAggregator(final String metricId, final DistributionBucket buckets) {
+    this(metricId, buckets, DEFAULT_CARDINALITY, DEFAULT_MAX_CARDINALITY);
+  }
+
+  public BasicHistogramAggregator(final String metricId, final DistributionBucket buckets, final int capacity) {
+    this(metricId, buckets, capacity, DEFAULT_MAX_CARDINALITY);
+  }
+
+  public BasicHistogramAggregator(final String metricId, final DistributionBucket buckets, final int capacity, final int maxCapacity) {
+    super(AGG_FIELDS.length + buckets.getCount(), capacity, maxCapacity, IDENTITY);
+    this.metricId = metricId;
+    this.buckets = buckets;
+    this.bucketCount = buckets.getCount();
+  }
+
+  @Override
+  public void apply(String[] tags, long latency, long timestamp) {
+    apply(tags, (int) latency, timestamp);
+  }
+
+  @Override
+  protected void combine(int[] table, long baseOffset, long value) {
+    addToAggField(table, baseOffset, 0, 1);
+    addToAggField(table, baseOffset, 1, value);
+    minAggField(table, baseOffset, 2, value);
+    maxAggField(table, baseOffset, 3, value);
+
+    //Increments the bucket counter by 1 responding to the given value
+    int bucketIndex = buckets.getBucketIndex(value);
+    addToDataField(table, baseOffset, AGG_FIELDS.length + bucketIndex, 1);
+  }
+
+  @Override
+  public Cursor cursor() {
+    return newCursor(false);
+  }
+
+  @Override
+  public Cursor sortedCursor() {
+    return newCursor(true);
+  }
+
+  private Cursor newCursor(boolean sorted) {
+    String[] fields = buildFields();
+    Type[] types = buildTypes(fields.length);
+    long[] identity = buildIdentity(fields.length);
+    return new CursorImpl(tagSets, fields, types, identity, sorted);
+  }
+
+  private long[] buildIdentity(int length) {
```

Comment:
The point is not clear. Could you please explain your intent?

Reply:
what is the point of this method?

Reply:
This is the identity values for the buckets.

Reply:
Yes, but why do you need this copy here? Its not modified anywhere?

Reply:
These are the identity values used by the Cursor to reset the values after reading. The record length depends on the number of buckets used for the histogram, I initialize this array once, and use it each time the Cursor object is created.

Reply:
Currently the Base aggregation like `count`, `sum`,`min`,`max` and buckets counts are tracked in the same table. Once we split that as part of https://github.com/ultrabrew/metrics/issues/18, we can default the identity values for the buckets.

Reply:
So do I understand this correctly: This method is intended to produce an array that contains copy of `IDENTITY` followed by `buckets.getCount()` zeroes?

If that is the case, please make it explicit in the code instead of relying on default initialized values. A comment explaining why would also help.